src/providers/Web3Provider.tsx

```diff
@@ -263,7 +275,7 @@ export const Web3ContextProvider: FC<PropsWithChildren> = ({ children }) => {
       throw new Error('[wRoseContract] not initialized!')
     }
 
-    return await wRoseContract.deposit({ value: amount, gasLimit: MAX_GAS_LIMIT, gasPrice: MAX_GAS_PRICE })
+    return await wRoseContract.deposit({ value: amount, gasLimit: MAX_GAS_LIMIT })
```

Comment:
Why is any of the new code needed? This makes a transaction without using the debounced gas price, and relies on metamask to estimate gas price

Reply:
There was a short discussion in the demo, when this was presented. @kostko mentioned that we should be using gas price from the network, instead of hardcoding it.

As for how calculation is done, it relies on provider calculation, it will calculate the gas price before sending it to MetaMask. At first I was thinking that I should just leave it empty to calculate the latest gas price before actually sending it to MetaMask, but you are right, there could be discrepancy with amount we are estimating and the one that is actually send to the provider.

Reply:
do we only need it for max-fee deposit?

Reply:
No, there is a label as well, showing an estimate. And upon switching from WROSE -> ROSE, if WROSE balance > ROSE balance = then it uses 100% of ROSE - fee and the 100% button - which I think you were referring to.

Reply:
okay. my preference is to remove the estimate display (leave it to metamask), and just do a hardcoded subtraction at 100% - fee (we discussed some -5x fee to let user do more transactions anyway)

Reply:
I mean, doing x5 fee seems unreasonable - as for the case where user wants to convert 100%(- fee) ROSE -> WROSE gets a warning message, and wallet can be refunded with ROSE. We can discuss this in next product meeting.

Reply:
untested simplified code: https://github.com/oasisprotocol/dapp-wrose/compare/ml/estimate-gas-price-via-provider...lw/no-estimate

Reply:
oh, i thought we decided to go with reserving extra for fees :shrug:

Reply:
Right, if we remove the label, the code gets simplified a lot. But since we are showing the label, I wouldn't take the auto calculation in wrap/unwrap, as it can cause discrepancy of what we show and the tx that would be submitted. 

I wouldn't hardcode it, in case $ROSE price rises in theory to 1000$, would mean you would be left with 10$ - actual fee. At least projects like Uniswap are doing it that way(100% conversion from ETH->WETH), why prevent users from doing an action, they can always top up the wallet.

Reply:
I would suggest, lets merge it the way it is. And we can discuss on product sync - added the topic.